python/graphstorm/dataloading/dataloading.py

```diff
@@ -87,7 +87,71 @@ def sample(self, seeds):
         input_nodes = {ntype: block.srcnodes[ntype].data[dgl.NID] for ntype in block.srctypes}
         return block, input_nodes
 
-class GSgnnEdgeDataLoader():
+class GSgnnEdgeDataLoaderBase():
+    """ The base dataloader class for edge tasks.
+
```

Comment:
Should users extend this class if they want to implement their own dataloader? If so, could you provide examples about what functions they need to overwrite to customize?

This may also apply to the `NodeDataLoaderBase` and `LinkPredictionDataLoaderBase`.

Reply:
are you sure docstring is a good place to write examples to illustrate how to customize a data loader?

Reply:
how complete the example should be?

Reply:
We expose this API mainly because we want users to use this Class to implement their own model. So it is better to give some explanation and example here.

I would suggest to give a sudo code to let users to know they need to extend this class.

Reply:
The methods that require users to implement are a common interface of Python iterator. any Python users should understand this interface. if we want to provide an example, we should provide a more realistic example.

Reply:
OK